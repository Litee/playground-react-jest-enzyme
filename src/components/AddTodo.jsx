import React, { Component } from 'react'

class AddTodo extends Component {
  constructor (props) {
    super(props)
    this.state = { value: '' }
    this.handleChange = e => {
      this.setState({ value: e.target.value })
    }
    this.handleSubmit = e => {
      this.props.onAddTodo(this.state.value)
      this.setState({ value: '' })
      e.preventDefault()
    }
  }

  render () {
    return (
      <form onSubmit={this.handleSubmit}>
        <label>Add todo:</label>
        <input
          type='text'
          value={this.state.value}
          onChange={this.handleChange}
        />
      </form>
    )
  }
}

export default AddTodo
