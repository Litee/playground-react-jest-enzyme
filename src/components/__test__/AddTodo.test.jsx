import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import AddTodo from '../AddTodo'

test('Must allow adding', () => {
  const onAddTodo = jest.fn()
  const wrapper = shallow(
    <AddTodo onAddTodo={onAddTodo} />
  )
  expect(shallowToJson(wrapper)).toMatchSnapshot()

  wrapper.find('input').simulate('change', {
    target: {value: 'ABCDE'}
  })
  wrapper.find('form').simulate('submit', {preventDefault: () => {}})
  expect(onAddTodo).toBeCalledWith('ABCDE')
})
