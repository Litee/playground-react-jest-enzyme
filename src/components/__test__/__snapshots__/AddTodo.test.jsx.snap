// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Must allow adding 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Add todo:
  </label>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
